File: packages/skillet-eval-python/src/skill_eval/utils/rate.py
from __future__ import annotations

from collections.abc import Callable
from typing import Literal

HIGH_THRESHOLD = 0.8
MID_THRESHOLD = 0.5

RateLevel = Literal["green", "yellow", "red"]

_COLOR_MAP: dict[RateLevel, str] = {
    "green": "green",
    "yellow": "yellow",
    "red": "red",
}


def rate_level(rate: float) -> RateLevel:
    if rate >= HIGH_THRESHOLD:
        return "green"
    if rate >= MID_THRESHOLD:
        return "yellow"
    return "red"


def rate_color(rate: float) -> Callable[[str], str]:
    tag = _COLOR_MAP[rate_level(rate)]

    def formatter(s: str) -> str:
        return f"[{tag}]{s}[/{tag}]"

    return formatter

File: packages/skillet/src/eval/utils/rate.ts
import pc from "picocolors";

export const HIGH_THRESHOLD = 0.8;
export const MID_THRESHOLD = 0.5;

export type RateLevel = "green" | "yellow" | "red";

export function rateLevel(rate: number): RateLevel {
	if (rate >= HIGH_THRESHOLD) return "green";
	if (rate >= MID_THRESHOLD) return "yellow";
	return "red";
}

const PC_COLORS = { green: pc.green, yellow: pc.yellow, red: pc.red } as const;

/** Return the picocolors formatter for a pass rate. */
export function rateColor(rate: number): (s: string) => string {
	return PC_COLORS[rateLevel(rate)];
}
